src/key.rs

```diff
@@ -168,9 +168,14 @@ impl str::FromStr for PublicKey {
     fn from_str(s: &str) -> Result<PublicKey, Error> {
         let mut res = [0u8; constants::UNCOMPRESSED_PUBLIC_KEY_SIZE];
         match from_hex(s, &mut res) {
-            Ok(constants::PUBLIC_KEY_SIZE) =>
-                PublicKey::from_slice(&res[0..constants::PUBLIC_KEY_SIZE]),
-            Ok(constants::UNCOMPRESSED_PUBLIC_KEY_SIZE) => PublicKey::from_slice(&res),
+            Ok(constants::PUBLIC_KEY_SIZE) => {
+                let bytes: [u8; constants::PUBLIC_KEY_SIZE] = res[0..constants::PUBLIC_KEY_SIZE]
+                    .try_into()
+                    .map_err(|_| Error::InvalidPublicKey)?;
```

Comment:
`try_from` only fails if length is wrong so we can just use `expect("length checked above")` or something similar here.

Reply:
I have changed it as suggested and ran just fmt to hopefully remove the CI fail